### PR TITLE
Sent button

### PR DESCRIPTION
**Pull request by MehdiElrafi**
Screenshot 2024-01-11 182015
image

### CODE REVIEW

**Comment by MehdiElrafi** [comment]
[Dans ordre de mission et NDF : Ajouter un bouton "Envoyer" à côté du bouton "Enregistrer" et "annuler"](https://trello.com/c/TXOyDN2K/289-dans-ordre-de-mission-et-ndf-ajouter-un-bouton-envoyer-%C3%A0-c%C3%B4t%C3%A9-du-bouton-enregistrer-et-annuler)

**Comment by MounirGaiby** [comment]
Can you change the size of "Supprimer" to become consistent with the other actions like "Envoyer" and "Modifier".

File: app/views/shared/requests/_request.html.erb [hunk 1]
```diff
@@ -38,6 +36,15 @@
...
+              <% if allowed_to?(:send_by_creator?, request) && request.aasm_state == "drafted" %>
+                <% next_state_path = request.is_a?(Expense) ? send(send_path, request, next_state: "create", type: @type) : send(send_path, request.user, request, next_state: "create", type: @type) %>
+                <%= button_to next_state_path, method: :patch, data: {turbo: false}, class: "text-gray-500 group flex items-center px-4 py-2 text-sm space-x-3 hover:text-indigo-600" do %>
+                  <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="h-5 w-5">
+                    <path stroke-linecap="round" stroke-linejoin="round" d="M15.59 14.37a6 6 0 01-5.84 7.38v-4.8m5.84-2.58a14.98 14.98 0 006.16-12.12A14.98 14.98 0 009.631 8.41m5.96 5.96a14.926 14.926 0 01-5.841 2.58m-.119-8.54a6 6 0 00-7.381 5.84h4.8m2.581-5.84a14.927 14.927 0 00-2.58 5.84m2.699 2.7c-.103.021-.207.041-.311.06a15.09 15.09 0 01-2.448-2.448 14.9 14.9 0 01.06-.312m-2.24 2.39a4.493 4.493 0 00-1.757 4.306 4.493 4.493 0 004.306-1.758M16.5 9a1.5 1.5 0 11-3 0 1.5 1.5 0 013 0z" />
+                  </svg>
+                  <span>Envoyer</span>
```
Comment (MounirGaiby) [inline app/views/shared/requests/_request.html.erb:45]: Why is this hard coded

File: app/services/handle_create_send_request_service.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class HandleCreateSendRequestService
+  def create_and_send(current_user, reaction, request)
+    if (request.is_a?(Expense) && request.amount > 0) || !request.is_a?(Expense)
+      if reaction == "createSend"
```
Comment (MounirGaiby) [inline app/services/handle_create_send_request_service.rb:4]: Why are you checking if reaction is "createSend" inside a create_and_send function. this should be outside this service in my opinion

File: app/views/time_requests/_form.html.erb [hunk 3]
```diff
@@ -50,6 +50,13 @@
     <button data-action="click->right#hide" type="button" class= "text-sm font-semibold leading-6 text-gray-900">
       <%= t("buttons.cancel") %>
     </button>
-    <%= form.submit t("buttons.save"), class: "rounded-md bg-indigo-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600" %>
+    <%= form.button :submit , name:"reaction", value: "create", class: "rounded-md bg-indigo-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600" do %>
+      <%= t("buttons.save") %>
+    <% end %>
+    <% if time_request.aasm_state == "drafted" %>
+      <%= form.button :submit , name:"reaction", value: "createSend", class: "rounded-md bg-indigo-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600" do %>
+        <%= t("buttons.save") %> && <%= t("buttons.send") %>
```
Comment (MounirGaiby) [inline app/views/time_requests/_form.html.erb:58]: Use "et" / "and" instead of &&. Because && could not be know to a general user

**Review by MounirGaiby — APPROVED** [review]
Maybe change reaction to action since it expresses more the intent.

**Pull request merged**